usaspending_api/common/logging.py

```diff
@@ -204,3 +222,77 @@ def ensure_logging(
             f"logger={cfg_logger_name}, formatter={cfg_formatter_name}, formatter_class={str(formatter_class)})"
         )
     return logger_to_use
+
+
+# CUSTOM Logging EXPORTER for debugging
+class LoggingSpanProcessor(SpanProcessor):
+    def on_end(self, span: ReadableSpan) -> None:
+        trace_id = span.context.trace_id
+        span_id = span.context.span_id
+        logger.debug(f"Span ended: trace_id={trace_id}, span_id={span_id}, {span.name}_attributes={span.attributes}")
+
+    def force_flush(self, timeout_millis: int = 30000) -> bool:
+        return True
+
+
+class CustomAttributeSpanProcessor(SpanProcessor):
+    def __init__(self, attribute_key, attribute_value):
+        self.attribute_key = attribute_key
+        self.attribute_value = attribute_value
+
+    def on_start(self, span, parent_context):
+        # Add the custom attribute when the span starts
+        span.set_attribute(self.attribute_key, self.attribute_value)
+
+    def on_end(self, span):
+        pass  # No action needed when the span ends
+
+
+def add_custom_attribute_span_processors(attribute_pairs: List[Tuple[str, str]]):
+    for key, value in attribute_pairs:
+        custom_attribute_span_processor = CustomAttributeSpanProcessor(key, value)
+        trace.get_tracer_provider().add_span_processor(custom_attribute_span_processor)
```

Comment:
A better interface could be to add a `tracer_provider` parameter to the function input parameters and have the calling code first call `get_tracer_provider` and pass it in.  That way there is more visibility in what is being modified by the function.  Another suggestion is that if this is just intended to be used with the function below (e.g. you want to discourage it from being imported into other files) you could make it "private" by prefixing it with an `_`.

Reply:
I went the route of passing in the tracerProvider in the function call.